### PR TITLE
Adding Beta-2 RN to the 5.4.0-BETA-2 branch

### PR DESCRIPTION
**Pull request by Serdaro**

### CODE REVIEW

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+* Fixed an issue where the Kinesis sink might loose data, when retrying on failures, during a terminal snapshot.
+https://github.com/hazelcast/hazelcast/pull/24779[#24779]
+* Fixed an issue where the member list was not updated after a cluster failover scenario.
+https://github.com/hazelcast/hazelcast/pull/24745[#24745]
+* Fixed an issue where the batches produced for write-behind queues were not having the expected size of entries.
+ https://github.com/hazelcast/hazelcast/issues/24763[#24763]
+* Fixed an issue where the fused Jet vertex was ignoring the configured local parallelism and using the default parallelism instead.
+https://github.com/hazelcast/hazelcast/issues/24683[#24683]
+* Fixed an issue where Hazelcast was sending empty map interceptor information to the members which are newly joined to the cluster; it was causing eager map initializations.
+https://github.com/hazelcast/hazelcast/pull/24639[#24639]
```
Comment (rebekah-lawrence) [inline docs/modules/release-notes/pages/5-4-0.adoc:290]: need to add something for anesoruk's Kafka partitions initial offsets contribution: https://github.com/hazelcast/hazelcast/pull/21546

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+* Implemented RAFT lifecycle interfaces for CPMap. #6800
+* Added the `getCPGroupIds()` method to the CP Subsystem API to allow listing all data structures within a CP group. #591
+* The "last update time" record of the map entries recovered from the disk persistence is not used anymore when calculating the entries' expiration times. #233
+* [.enterprise]*Enterprise* Added the ability to check whether the `ADVANCED_CP` feature is present in the Hazelcast Platform Enterprise license, to enable CPMap. #157
+
+=== [.enterprise]*Enterprise* WAN Replication
+
+* Improved dead connection handling for WAN replication static IP discovery, by introducing health check to the discovery strategy.
+https://github.com/hazelcast/hazelcast/pull/25364[#25364]
+* Added support for the dynamic adding of WAN replication configurations using Java API.
```
[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/release-notes/pages/5-4-0.adoc:139]: It depends on context for me - normally singular is right but if this feature enables dynamically adding different configurations then it's fine. Are there instances where you would have different configurations for WAN replication?

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 3]
```diff
@@ -0,0 +1,335 @@
+= 5.4.0-BETA-2 Release Notes
+
+CAUTION: Starting with this release of Hazelcast Platform, the minimum supported Java version is 17.
+
+== New Features
+
+* **User Code Namespaces**: Enable deployment, undeployment and redeployment of your custom classes.
```
[the remark this answers is not in the document]
Comment (Serdaro) [reply docs/modules/release-notes/pages/5-4-0.adoc:7]: I don't see it is merged (/clusters/user-code-namespaces.adoc). We can always direct the users when the content is there. When it is merged, we can cross-reference it to the "Removed/Deprecated Features" section, where we mention UCD is deprecated.

File: docs/modules/release-notes/pages/5-4-0.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (Serdaro) [reply docs/modules/release-notes/pages/5-4-0.adoc:139]: This is not related to configuring multiple files, but configuring multiple elements "in the configuration file".

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+https://github.com/hazelcast/hazelcast/pull/24928[#24928]
+* Fixed an issue where the lite members were not reporting statistics for map operations.
+https://github.com/hazelcast/hazelcast/pull/24871[#24871]
+* Fixed an issue where the blacklisting was being ignored after a split-brain happens.
+https://github.com/hazelcast/hazelcast/pull/24830[#24830]
+* Fixed an issue where the Kinesis sink might loose data, when retrying on failures, during a terminal snapshot.
+https://github.com/hazelcast/hazelcast/pull/24779[#24779]
+* Fixed an issue where the member list was not updated after a cluster failover scenario.
+https://github.com/hazelcast/hazelcast/pull/24745[#24745]
+* Fixed an issue where the batches produced for write-behind queues were not having the expected size of entries.
```
[the remark this answers is not in the document]
Comment (Serdaro) [reply docs/modules/release-notes/pages/5-4-0.adoc:285]: size means number of entries within this context.

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+ https://github.com/hazelcast/hazelcast/issues/24763[#24763]
+* Fixed an issue where the fused Jet vertex was ignoring the configured local parallelism and using the default parallelism instead.
+https://github.com/hazelcast/hazelcast/issues/24683[#24683]
+* Fixed an issue where Hazelcast was sending empty map interceptor information to the members which are newly joined to the cluster; it was causing eager map initializations.
```
[the remark this answers is not in the document]
Comment (Serdaro) [reply docs/modules/release-notes/pages/5-4-0.adoc:289]: I saw you changed "that" to "which" in some other former PRs :).  Any when-to-use-what standardization on this?

File: docs/modules/release-notes/pages/5-4-0.adoc (hunk 1 above)
Comment (Serdaro) [reply docs/modules/release-notes/pages/5-4-0.adoc:290]: @rebekah-lawrence , the related PR is still open, and I agree with https://github.com/hazelcast/hz-docs/pull/996/files#r1479912945

Comment (Serdaro) [reply docs/modules/release-notes/pages/5-4-0.adoc:292]: https://github.com/hazelcast/hz-docs/pull/1001/commits/479bf5651dc132c047124c3fd8277573cca8be97

File: docs/modules/release-notes/pages/5-4-0.adoc [hunk 6]
```diff
@@ -0,0 +1,339 @@
...
+* The connector for Elasticsearch 6 is removed, as the Elasticsearch 6 module is removed from Hazelcast distributions.
+https://github.com/hazelcast/hazelcast/pull/24734[#24734]
+* The evaluation tool for IMDG 3.x users (Hazelcast 3 Connector) is removed. In the upcoming releases, a new tooling for migrating data from 3.x versions will be introduced.
+https://github.com/hazelcast/hazelcast/pull/25051[#25051]
+* Transactions are deprecated, will be removed as of Hazelcast Platform version 7.0.
+** Only maintenance support will be provided for transactions.
+** Transactions will no longer be actively developed.
+** Issues arising from new or existing implementations of transactions will no longer be supported.
++
+In the meantime, if you are already using transactions and want to discuss alternatives, please get in touch with xref:getting-started:support.adoc[Hazelcast Support].
```
[the remark this answers is not in the document]
Comment (Serdaro) [reply docs/modules/release-notes/pages/5-4-0.adoc:319]: We will clarify this with Avtar this week

[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/release-notes/pages/5-4-0.adoc:319]: I'm thinking the line might need revised/clarified or even removed tbh:
> Any issues arising from new or existing implementations of transactions will not be supported.
The sentences either side of it probably cover it sufficiently.

[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/release-notes/pages/5-4-0.adoc:319]: Revising my above suggested change to remove that line ^

[the remark this answers is not in the document]
Comment (oliverhowell) [reply docs/modules/release-notes/pages/5-4-0.adoc:319]: Alt suggestion is just to use what was put into main docs:
```suggestion
Transactions have been deprecated, and will be removed as of Hazelcast version 7.0. An improved version of this feature is under consideration. If you are already using transactions, get in touch and share your use case. Your feedback will help us to develop a solution that meets your needs.
```

**Pull request merged**